frontend/src/components/core/Link.tsx

```diff
@@ -15,8 +15,22 @@ const createButtonLink = (Button:any, props:any) => {
   return <Link to={to}>{button}</Link>;
 };
 
+// Wrap a button inside of a Link that triggers an onClick function
+const createButtonOnClick = (Button:any, props:any) => {
+  const { onClickFunc, children, ...rest } = props;
+  const button = (
+    <Button onClick={onClickFunc} {...rest}>
+      {children}
+    </Button>
+  );
+
+  return button;
+};
+
 export const PrimaryButtonLink = (props:any) => createButtonLink(PrimaryButton, props);
 
+export const SocketButtonConnection = (props:any) => createButtonOnClick(PrimaryButton, props);
+
```

Comment:
nit: I think you can get away without this and just do something like this in the `components/core/Button.tsx` folder, since there isn't actually any Link wrapping involved:

```js
export const SocketButton = styled.button`
  // styles go here
`;
```

Reply:
I realized that if I'm just using the `PrimaryButton` style, there wasn't really a need for me to create all those other button types and such -- I just changed the PR to use a `PrimaryButton` directly in the `Join.tsx` page. (This is especially true because eventually these buttons won't be there, so creating an extra style for this task is superfluous work.)